src/main/java/uk/gov/fco/documentupload/service/ocr/OCRService.java

```diff
@@ -0,0 +1,134 @@
+package uk.gov.fco.documentupload.service.ocr;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.SdkField;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.rekognition.model.*;
+import software.amazon.awssdk.services.textract.TextractClient;
+import software.amazon.awssdk.services.textract.model.*;
+import uk.gov.fco.documentupload.service.storage.FileUpload;
+
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+import java.util.*;
+
+import software.amazon.awssdk.services.rekognition.RekognitionClient;
+
+@Service
+@Slf4j
+public class OCRService {
+    private boolean enabled;
+
+    private int sharpnessThreshold;
+
+    private RekognitionClient rekognition;
+
+    private TextractClient textract;
+
+    @Autowired
+    public OCRService(@Value("${ocr.enabled}") @NotNull boolean enabled, @Value("${ocr.sharpness.threshold}") int sharpnessThreshold) {
+        this.enabled = enabled;
+        this.sharpnessThreshold = sharpnessThreshold;
+        rekognition = RekognitionClient.builder().region(Region.EU_WEST_2).credentialsProvider(EnvironmentVariableCredentialsProvider.create()).build();
+        textract = TextractClient.builder().region(Region.EU_WEST_2).credentialsProvider(EnvironmentVariableCredentialsProvider.create()).build();
+    }
+
+    public boolean passesQualityCheck(FileUpload upload) throws IOException {
+        if (enabled) {
+            log.info("Starting image quality check");
+            SdkBytes bytes = SdkBytes.fromInputStream(upload.getInputStream());
+            Image image = Image.builder().bytes(bytes).build();
+            DetectLabelsRequest request = DetectLabelsRequest.builder().image(image).featuresWithStrings("IMAGE_PROPERTIES").build();
+            DetectLabelsImageProperties results = rekognition.detectLabels(request).imageProperties();
+            return results.quality().sharpness() >= sharpnessThreshold;
+        }
+        return true;
+    }
+
+    public String extractData(FileUpload upload) throws IOException {
```

Comment:
maybe actually we'll just remove this block. you can keep it on another branch somewhere or just use git history to get it